module/evdi_drv.c

```diff
@@ -114,11 +116,44 @@ static void evdi_add_device(void)
 	evdi_context.dev_count++;
 }
 
+#if KERNEL_VERSION(4, 12, 0) >= LINUX_VERSION_CODE
+static int evdi_platform_probe(struct platform_device *pdev)
+{
+	struct drm_device *dev;
+	int ret;
+
+	EVDI_CHECKPT();
+
+	dev = drm_dev_alloc(&driver, &pdev->dev);
+	if (IS_ERR(dev))
+		return PTR_ERR(dev);
+
+	ret = evdi_driver_setup_early(dev);
+	if (ret)
+		goto err_free;
+
+	// drm_set_unique?
+
+	ret = drm_dev_register(dev, 0);
+	if (ret)
+		goto err_free;
+
+	evdi_stats_init(dev->dev_private);
+
+	return 0;
+
+err_free:
+	drm_dev_unref(dev);
+	return ret;
+}
+#else
+/* old-style load callback */
```

Comment:
We are not using comments to denote what is ```#else``` for. 
If you would put the short version above the long version the ```#if <condition> #else``` will be close to each other and it will be obvious what ```#else``` is.